### PR TITLE
i18n(fr): update `content-collections.mdx` & `modules/astro-content.mdx`

### PR DESCRIPTION
**Pull request by ArmandPhilippot**
#### Description (required)

* Adds changes from #9240, #10143, #10209 and #10213 to the French translations of `guides/content-collections.mdx` and `reference/modules/astro-content.mdx`.
* Adds back the `ReadMore` in `guides/markdown-content.mdx` (see https://github.com/withastro/docs/pull/10319#discussion_r1876377245)
* Adds back the link in `integrations-guide/mdx.mdx` (see #10328)
* Fix malformed code snippet in `integrations-guide/mdx.mdx` (see #10344)
* Adds missing update in the Unit 6-3 of the tutorial

#### Related issues & labels (optional)

- Suggested label: i18n

### CODE REVIEW

**Review by thomasbnt — APPROVED** [review]
LGTM

File: src/content/docs/fr/guides/content-collections.mdx [hunk 1]
```diff
@@ -1,240 +1,263 @@
...
-:::
+Le chargeur `file()` crée plusieurs entrées à partir d'un seul fichier local. Chaque entrée du fichier doit avoir une propriété de clé `id` unique. Il accepte un chemin de fichier `base` vers votre fichier et éventuellement une [fonction `parser`](#fonction-parser) pour les fichiers de données qu'il ne peut pas analyser automatiquement. Utilisez ce chargeur lorsque votre fichier de données peut être analysé comme un tableau d'objets.
 
-Pour tirer le meilleur parti de vos collections de contenus, créez un fichier `src/content/config.ts` dans votre projet (les extensions `.js` et `.mjs` sont également supportées.) Il s'agit d'un fichier spécial qu'Astro chargera et utilisera automatiquement pour configurer vos collections de contenus.
+```ts  title="src/content.config.ts" {5,9}
+import { defineCollection, z } from 'astro:content';
+import { glob, file } from 'astro/loaders'; // Non disponible avec l'API héritée
 
+const blog = defineCollection({
+  loader: glob({ pattern: "**\/*.md", base: "./src/data/blog" }),
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/content-collections.mdx:97]:
```suggestion
  loader: glob({ pattern: "**/*.md", base: "./src/data/blog" }),
```

File: src/content/docs/fr/guides/content-collections.mdx [hunk 2]
```diff
@@ -1,240 +1,263 @@
...
+Les schémas alimentent également les typages automatiques TypeScript d'Astro pour votre contenu. Lorsque vous définissez un schéma pour votre collection, Astro génère et applique automatiquement une interface TypeScript. Le résultat est une prise en charge complète de TypeScript lorsque vous interrogez votre collection, y compris la saisie semi-automatique des propriétés et la vérification des types.
+
+Chaque frontmatter ou propriété de données de vos entrées de collection doit être défini à l’aide d’un type de données Zod :
+
+```ts title="src/content.config.ts" {6-11,15-19}
+import { defineCollection, z } from 'astro:content';
+import { glob, file } from 'astro/loaders'; // Non disponible avec l'API héritée
+
+const blog = defineCollection({
+  loader: glob({ pattern: "**\/*.md", base: "./src/data/blog" }),
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/content-collections.mdx:233]:
```suggestion
  loader: glob({ pattern: "**/*.md", base: "./src/data/blog" }),
```

File: src/content/docs/fr/guides/content-collections.mdx [hunk 3]
```diff
@@ -246,47 +269,53 @@ defineCollection({
...
+Avec la fonction `reference()` de l'API Collections, vous pouvez définir une propriété dans un schéma de collection en tant qu'entrée d'une autre collection. Par exemple, vous pouvez exiger que chaque entrée `space-shuttle` inclue une propriété `pilot` qui utilise le schéma propre à la collection `pilot` pour la vérification du type, la saisie semi-automatique et la validation.
+
+Un exemple courant est un article de blog qui fait référence à des profils d'auteur réutilisables stockés au format JSON ou à des URL d'articles associés stockées dans la même collection :
+
+```ts title="src/content.config.ts"
 import { defineCollection, reference, z } from 'astro:content';
 
 const blog = defineCollection({
-  type: 'content',
+  loader: glob({ pattern: '**\/[^_]*.md', base: "./src/data/blog" }),
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/content-collections.mdx:307]:
```suggestion
  loader: glob({ pattern: '**/[^_]*.md', base: "./src/data/blog" }),
```

File: src/content/docs/fr/guides/content-collections.mdx [hunk 4]
```diff
@@ -246,47 +269,53 @@ defineCollection({
...
     author: reference('authors'),
-    // Référence un tableau d'articles liés de la collection `blog` par `slug`id`.
+    // Faire référence à un tableau d'articles connexes de la collection `blog` par `slug`
     relatedPosts: z.array(reference('blog')),
   })
 });
 
 const authors = defineCollection({
-  type: 'data',
+  loader: glob({ pattern: '**\/[^_]*.json', base: "./src/data/authors" }),
```
Comment (ArmandPhilippot) [inline src/content/docs/fr/guides/content-collections.mdx:318]:
```suggestion
  loader: glob({ pattern: '**/[^_]*.json', base: "./src/data/authors" }),
```

**Review by ArmandPhilippot — COMMENTED** [review]
Apply changes from #10209 and #10213 (the last one is not yet validated but... since this is the same thing I guess it's safe to apply it now)

**Review by thomasbnt — APPROVED** [review]
LGTM!

**Pull request merged**